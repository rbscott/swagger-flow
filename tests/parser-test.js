import yaml from 'js-yaml';

import parse from '../src/parser';

describe('parse', () => {
  describe('primitives', () => {
    it('Should parse an integer primitives', () => {
      const input = yaml.safeLoad(`
definitions:
  basicInteger:
    type: integer
`);
      expect(parse(input).items).toMatchSnapshot();
    });

    it('Should parse an number primitives', () => {
      const input = yaml.safeLoad(`
definitions:
  basicNumber:
    type: number
`);
      expect(parse(input).items).toMatchSnapshot();
    });

    it('Should parse a string primitives', () => {
      const input = yaml.safeLoad(`
definitions:
  basicString:
    type: string
`);
      expect(parse(input).items).toMatchSnapshot();
    });

    it('Should parse an boolean primitives', () => {
      const input = yaml.safeLoad(`
definitions:
  basicBoolean:
    type: boolean
`);
      expect(parse(input).items).toMatchSnapshot();
    });

  });

  describe('enum', () => {
    it('Should parse a basic enum', () => {
      const input = yaml.safeLoad(`
definitions:
  basicEnum:
    type: string
    enum:
      - value1
      - value2
      - value3
`);
      expect(parse(input).items).toMatchSnapshot();
    });
  });

  describe('array', () => {
    it('Should parse a basic array', () => {
      const input = yaml.safeLoad(`
definitions:
  basicArray:
    type: array
    items:
      type: string
`);
      expect(parse(input).items).toMatchSnapshot();
    });

    it('Should parse an additional items boolean', () => {
      const input = yaml.safeLoad(`
definitions:
  basicArray:
    type: array
    items:
      type: string
    additionalItems: true
`);
      expect(parse(input).items).toMatchSnapshot();
    });

    it('Should parse an additional items boolean (false)', () => {
      const input = yaml.safeLoad(`
definitions:
  basicArray:
    type: array
    items:
      type: string
    additionalItems: false
`);
      expect(parse(input).items).toMatchSnapshot();
    });

    it('Should parse an additional items schema', () => {
      const input = yaml.safeLoad(`
definitions:
  basicArray:
    type: array
    items:
      type: string
    additionalItems:
      type: number
`);
      expect(parse(input).items).toMatchSnapshot();
    });

    it('Should parse arrays of arrays.', () => {
      const input = yaml.safeLoad(`
definitions:
  basicArray:
    type: array
    items:
      type: array
      items:
        type: string
`);
      expect(parse(input).items).toMatchSnapshot();
    });

  });
});
