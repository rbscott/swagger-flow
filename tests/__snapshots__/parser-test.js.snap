// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse array Should parse a basic array 1`] = `
Array [
  Object {
    "additionalItems": undefined,
    "id": "definitions/basicArray",
    "items": Object {
      "id": "definitions/basicArray/ArrayType",
      "name": "basicArray_ArrayType",
      "type": "string",
    },
    "name": "basicArray",
    "type": "array",
  },
]
`;

exports[`parse array Should parse an additional items boolean (false) 1`] = `
Array [
  Object {
    "additionalItems": false,
    "id": "definitions/basicArray",
    "items": Object {
      "id": "definitions/basicArray/ArrayType",
      "name": "basicArray_ArrayType",
      "type": "string",
    },
    "name": "basicArray",
    "type": "array",
  },
]
`;

exports[`parse array Should parse an additional items boolean 1`] = `
Array [
  Object {
    "additionalItems": true,
    "id": "definitions/basicArray",
    "items": Object {
      "id": "definitions/basicArray/ArrayType",
      "name": "basicArray_ArrayType",
      "type": "string",
    },
    "name": "basicArray",
    "type": "array",
  },
]
`;

exports[`parse array Should parse an additional items schema 1`] = `
Array [
  Object {
    "additionalItems": Object {
      "id": "definitions/basicArray/AdditionalArrayType",
      "name": "basicArray_AdditionalArrayType",
      "type": "integer",
    },
    "id": "definitions/basicArray",
    "items": Object {
      "id": "definitions/basicArray/ArrayType",
      "name": "basicArray_ArrayType",
      "type": "string",
    },
    "name": "basicArray",
    "type": "array",
  },
]
`;

exports[`parse array Should parse arrays of arrays. 1`] = `
Array [
  Object {
    "additionalItems": undefined,
    "id": "definitions/basicArray",
    "items": Object {
      "additionalItems": undefined,
      "id": "definitions/basicArray/ArrayType",
      "items": Object {
        "id": "definitions/basicArray/ArrayType/ArrayType",
        "name": "basicArray_ArrayType_ArrayType",
        "type": "string",
      },
      "name": "basicArray_ArrayType",
      "type": "array",
    },
    "name": "basicArray",
    "type": "array",
  },
]
`;

exports[`parse enum Should parse a basic enum 1`] = `
Array [
  Object {
    "id": "definitions/basicEnum",
    "name": "basicEnum",
    "type": "enum",
    "values": Array [
      "value1",
      "value2",
      "value3",
    ],
  },
]
`;

exports[`parse primitives Should parse a string primitives 1`] = `
Array [
  Object {
    "id": "definitions/basicString",
    "name": "basicString",
    "type": "string",
  },
]
`;

exports[`parse primitives Should parse an boolean primitives 1`] = `
Array [
  Object {
    "id": "definitions/basicBoolean",
    "name": "basicBoolean",
    "type": "boolean",
  },
]
`;

exports[`parse primitives Should parse an integer primitives 1`] = `
Array [
  Object {
    "id": "definitions/basicInteger",
    "name": "basicInteger",
    "type": "integer",
  },
]
`;

exports[`parse primitives Should parse an number primitives 1`] = `
Array [
  Object {
    "id": "definitions/basicNumber",
    "name": "basicNumber",
    "type": "integer",
  },
]
`;
